storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+        for (int i = 0; i < toks.size(); i++) {
+            output = output + "/" + toks.get(i);
+        }
+        return output;
+    }
+    public static String normalizePath (String path) {
+        return toksToPath(tokenizePath(path));
+    }
+
+    public static void exitProcess (int val, Object... msg) {
+        StringBuilder errorMessage = new StringBuilder();
+        errorMessage.append("halting process: ");
+        for (Object oneMessage: msg) {
+            errorMessage.append(oneMessage);
+        }
+        String combinedErrorMessage = errorMessage.toString();
+        LOG.error(combinedErrorMessage, new RuntimeException(combinedErrorMessage));
+        Runtime.getRuntime().exit(val);
+    }
+
+    public static Object defaulted(Object val, Object defaultObj) {
+        if (val != null) {
+            return val;
+        } else {
+            return defaultObj;
+        }
+    }
+
+    /**
+     * "{:a 1 :b 1 :c 2} -> {1 [:a :b] 2 :c}"
+     *
+     * Example usage in java:
+     *  Map<Integer, String> tasks;
+     *  Map<String, List<Integer>> componentTasks = Utils.reverse_map(tasks);
+     *
+     * @param map
+     * @return
+     */
+    public static <K, V> HashMap<V, List<K>> reverseMap(Map<K, V> map) {
+        HashMap<V, List<K>> rtn = new HashMap<V, List<K>>();
+        if (map == null) {
+            return rtn;
+        }
+        for (Entry<K, V> entry : map.entrySet()) {
+            K key = entry.getKey();
+            V val = entry.getValue();
+            List<K> list = rtn.get(val);
+            if (list == null) {
+                list = new ArrayList<K>();
+                rtn.put(entry.getValue(), list);
+            }
+            list.add(key);
+        }
+        return rtn;
+    }
+
+    /**
+     * "{:a 1 :b 1 :c 2} -> {1 [:a :b] 2 :c}"
+     *
+     */
+    public static HashMap reverseMap(List listSeq) {
+        HashMap<Object, List<Object>> rtn = new HashMap();
+        if (listSeq == null) {
+            return rtn;
+        }
+        for (Object entry : listSeq) {
+            List listEntry = (List) entry;
+            Object key = listEntry.get(0);
+            Object val = listEntry.get(1);
+            List list = rtn.get(val);
+            if (list == null) {
+                list = new ArrayList<Object>();
+                rtn.put(val, list);
+            }
+            list.add(key);
+        }
+        return rtn;
+    }
+
+
+    /**
+     * Gets the pid of this JVM, because Java doesn't provide a real way to do this.
+     *
+     * @return
+     */
+    public static String processPid() throws RuntimeException {
+        String name = ManagementFactory.getRuntimeMXBean().getName();
+        String[] split = name.split("@");
+        if (split.length != 2) {
+            throw new RuntimeException("Got unexpected process name: " + name);
+        }
+        return split[0];
+    }
+
+    public static int execCommand(String command) throws ExecuteException, IOException {
```

Comment:
Please please file a JIRA to make this be `String... command` instead.  We should not be splitting things on space, especially in some places where we join on space to send to this so we can split on space again.  Come on why are we doing this???